src/supervisor.adoc

```diff
@@ -2270,6 +2279,45 @@ Invalid PTEs using a bounded timer, or making address-translation caches
 coherent with store instructions that modify PTEs.
 ====
 
+[[sec:svukte]]
+== "Svukte" Extension for Address-Independent Latency of User-Mode Faults to Supervisor Addresses, Version 0.3
+
+The Svukte extension provides a means to make user-mode accesses to supervisor
+memory raise page faults in constant time, mitigating attacks that attempt to
```

Comment:
PR feedback from tech chairs: time invariant based on page table contents, not constant time of the whole exception handling.
```suggestion
memory raise page faults with timing independent of page table data, mitigating attacks that attempt to
```

Reply:
I think the point is to not even look at the page table contents. The very act of accessing the page table will make it not constant time and could expose KASLR.

Reply:
I'm certainly open to other ways of phrasing this, but I agree with Ved that the proposed wording seems to unintentionally weaken the spec by allowing implementations to perform the first level of the page-table walk then discard the result.  (But I think the proposed wording does prohibit accessing subsequent levels of the page table.)

Unlike Ved, I haven't convinced myself that opens up a timing attack, but even so, it wasn't my intent.

Another option would be "with timing independent of the contents of `satp` or the page tables".

Another possibility is "in constant time" => "without accessing the page tables", but that has the downside of shifting the emphasis towards the implementation and away from the high-level intent.

Reply:
Ooh I like that one: "with timing independent of the contents of satp or the page tables".

Reply:
Me too, but only if we can find a way to phrase it that includes all active `atp` registers.  "independent of the address-translation configuration and the page tables"?

Reply:
Maybe we could leave the normative statement as is and add a short note to guide implementation with some text like "without accessing page tables or address translation caches" so that it provides enough hints that this fault is supposed to be a fault caused by agen, similar to an invalid VA or too-wide GPA case, and not a TLB/PWH caused fault?

Reply:
Maybe we could leave the normative statement as is and add a short note to guide implementation with some text like "without accessing page tables or address translation caches" so that it provides enough hints that this fault is supposed to be a fault caused by agen, similar to an invalid VA or too-wide GPA case, and not a TLB/PWH caused fault?

Reply:
I can live with either my most recent proposal or Ved's suggestion.  Let's see if anyone else chimes in.

Reply:
I went with my version, but we can revisit this later if need be.